app/src/main/java/com/dimelo/sampleapp/MainActivity.java

```diff
@@ -68,11 +74,28 @@ else if (error.statusCode == DimeloConnection.DimeloError.TIMEOUT_ERROR){
         }
     };
 
-    private void setupDimelo(){
+    private void setupDimelo() {
         String secret = BuildConfig.DIMELO_SDK_SECRET; //edit in gradle.properties
         Dimelo.setup(this);
         Dimelo dimelo = Dimelo.getInstance();
         dimelo.setApiSecret(secret);
+//        dimelo.setUserIdentifier("42");
+        dimelo.setUserName("John Doe");
+
+        JSONObject authInfo = new JSONObject();
+        PackageInfo pInfo = null;
+        try {
+            pInfo = getPackageManager().getPackageInfo(getPackageName(), 0);
```

Comment:
Well I'd prefer a less convoluted example as this is a sample app (no package info).

But we should add that in the header of our requests.